### PR TITLE
Fixing validation on multiple choice questions

### PR DESCRIPTION
**Pull request by mehtadarshan1**

### CODE REVIEW

File: server/questions.js [hunk 1]
```diff
@@ -52,12 +52,17 @@ exports.addQuestionByType = function(qType, question, callback) {
...
 
 		default:
-			//callback('type is not defined', null);
-			questionToAdd.type = common.questionTypes.REGULAR.value;
+			return callback({status:400, msg:'Type of Question is Undefined'}, null)
+	}
+
+	// validate question by its type
+	var result = validateQuestionByType(questionToAdd, questionToAdd.type);
+	if (result != null){
```
Comment (esteeroramy) [inline server/questions.js]: javascript not equal is `!==`

File: server/questions.js [hunk 2]
```diff
@@ -96,7 +101,48 @@ exports.updateQuestionById = function(questionId, info, callback) {
...
 
 var updateQuestionById = function(questionId, info, callback) {
-	db.updateQuestionById(questionId, info, callback);
+	updateQuestionByType(questionId,info,callback);
+}
+
+var updateQuestionByType = function(qId, infoToUpdate, callback){
+	// Get Type of question and validate it
+	db.lookupQuestionById(qId, function(err, question){
+		if(err){
```
Comment (esteeroramy) [inline server/questions.js:110]: space after if

File: server/questions.js [hunk 3]
```diff
@@ -96,7 +101,48 @@ exports.updateQuestionById = function(questionId, info, callback) {
...
+			return callback({status:500, msg:err},null);
+		}
+
+		// validate question by its type
+		var result = validateQuestionByType(infoToUpdate, question.type);
+		if (result != null){
```
Comment (esteeroramy) [inline server/questions.js]: `!==`

**Review by esteeroramy — APPROVED** [review]

File: server/questions.js [hunk 4]
```diff
@@ -52,12 +52,17 @@ exports.addQuestionByType = function(qType, question, callback) {
...
 
 		default:
-			//callback('type is not defined', null);
-			questionToAdd.type = common.questionTypes.REGULAR.value;
+			return callback({status:400, msg:'Type of Question is Undefined'}, null)
+	}
+
+	// validate question by its type
+	var result = validateQuestionByType(questionToAdd, questionToAdd.type);
+	if (result !== null){
```
Comment (ahmedqarmout2) [inline server/questions.js:64]: if (result)

File: server/questions.js [hunk 5]
```diff
@@ -52,12 +52,17 @@ exports.addQuestionByType = function(qType, question, callback) {
 
 		case common.questionTypes.MULTIPLECHOICE.value:
 			questionToAdd.type = common.questionTypes.MULTIPLECHOICE.value;
-			questionToAdd.choices = question.choices;
+			questionToAdd.choices = question.choices ? question.choices : [];
```
Comment (ahmedqarmout2) [inline server/questions.js:55]: validate at the beginning of the function. Then you wont need to check question.choice

File: server/questions.js (hunk 4 above)
Comment (esteeroramy) [reply server/questions.js:64]: you mean `if(!result)`?

File: server/questions.js [hunk 6]
```diff
@@ -96,7 +101,48 @@ exports.updateQuestionById = function(questionId, info, callback) {
 }
 
 var updateQuestionById = function(questionId, info, callback) {
-	db.updateQuestionById(questionId, info, callback);
+	updateQuestionByType(questionId,info,callback);
```
Comment (ahmedqarmout2) [inline server/questions.js:104]: updateQuestionByType does not make sense, there are multiple questions of the same type. Lets put the content of updateQuestionByType in this function. That way when you do update question by id , you will do the validation then pass to db.udpateQuestionById

File: server/questions.js [hunk 7]
```diff
@@ -96,7 +101,48 @@ exports.updateQuestionById = function(questionId, info, callback) {
...
+		// validate question by its type
+		var result = validateQuestionByType(infoToUpdate, question.type);
+		if (result !== null){
+			return callback({status:400, msg:result}, null)
+		}
+		db.updateQuestionById(qId, infoToUpdate, callback);
+	});
+}
+
+var validateQuestionByType = function(question, type){
```
Comment (ahmedqarmout2) [inline server/questions.js:123]: We need to validate the basics here, title, body, points etc.

File: server/questions.js (hunk 4 above)
Comment (esteeroramy) [reply server/questions.js:64]: nvm ignore my comment

File: server/questions.js (hunk 5 above)
Comment (mehtadarshan1) [reply server/questions.js:55]: I have to check for it in the question type it self, thats why I have those switch statements

Comment (ahmedqarmout2) [reply server/questions.js:55]: But the validation will take care of checking for you

File: server/questions.js (hunk 7 above)
Comment (mehtadarshan1) [reply server/questions.js:123]: Yes that is a good point,

File: server/questions.js (hunk 6 above)
Comment (mehtadarshan1) [reply server/questions.js:104]: Yeah we can combine it into a single function, so no confusion

File: server/questions.js (hunk 5 above)
Comment (mehtadarshan1) [reply server/questions.js:55]: right... we can have all the validation done in that one function

**Pull request merged**